src/hooks/index.js

```diff
@@ -337,6 +337,69 @@ exports.restrictToUsersOwnCourses = hook => {
 	});
 };
 
+exports.restrictToUsersOwnLessons = hook => {
+	let userService = hook.app.service('users');
+	return userService.find({
+		query: {
+			_id: hook.params.account.userId,
+			$populate: 'roles'
+		}
+	}).then(res => {
+		let access = false;
+		res.data[0].roles.map(role => {
+			if (role.name === 'administrator' || role.name === 'superhero' )
+				access = true;
+		});
+		if (access)
+			return hook;
+
+		// before-hook
+		if(hook.type === "before") {
+			let populate = hook.params.query.$populate;
+			if (typeof(populate) === 'undefined') {
+				populate = ['courseId'];
+			} else if (Array.isArray(populate) && !populate.includes("courseId")) {
+				populate.push('courseId');
+			}
```

Comment:
How should it work if input $populate is not an array?

Reply:
$populate in query should always be an array or undefined, or at least we don't use it an other way. If so, however, I don't touch it.